Lib/subprocess.py

```diff
@@ -1494,6 +1494,8 @@ def _execute_child(self, args, executable, preexec_fn, close_fds,
                             raise FileNotFoundError('shell not found: neither %ComSpec% nor %SystemRoot% is set')
                     if os.path.isabs(comspec):
                         executable = comspec
+                else:
+                    comspec = executable
 
                 args = '{} /c "{}"'.format (comspec, args)
```

Comment:
The value of `args` should quote `comspec` because the path may contain spaces. Otherwise the shell process won't parse its command line correctly.

Reply:
That was a problem before too, so let's make it a new issue. Assuming the backports for the original change go it, it'll automatically flow back smoothly enough (whereas this PR may not).